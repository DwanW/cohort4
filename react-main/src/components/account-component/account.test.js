import React from 'react';
import { shallow } from 'enzyme';
import Account from './account';

describe('Account Component test', () => {
    let wrapper;
    const mockName = "house";
    const mockBalance = 350;
    let mockOnDelete;
    let mockOnValueChange;
    const mockIdx = 3;
    const mockKey = "3house";

    beforeEach(()=> {
        mockOnDelete = jest.fn();
        mockOnValueChange = jest.fn();

        const mockProps = {
            key: mockKey,
            idx: mockIdx,
            name: mockName,
            balance:mockBalance,
            onDelete: mockOnDelete,
            onValueChange: mockOnValueChange
        }

        wrapper = shallow(<Account {...mockProps}/>)
    })
    
    it('should render Account Component', ()=> {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.state().transaction).toBe(0);
    });

    it('should render the correct name and balance', ()=> {
        expect(wrapper.find('.accHeader').text()).toBe(mockName);
        expect(wrapper.find('.acc-result').text()).toBe(`Account ${mockName} total balance: $${mockBalance}`);
    });

    it('should call onDelete and onValueChange when clicked', ()=>{
        wrapper.find('.acc-delete').simulate('click');
        expect(mockOnDelete).toHaveBeenCalled();
        wrapper.find('.deposit').simulate('click');
        expect(mockOnValueChange).toHaveBeenCalled();
    });

    it('should render correct controlled input value', () => {
        wrapper.find('.acc-inputbox').simulate('change', {target: {value: '100'}});
        expect(wrapper.find('.acc-inputbox').props().value).toEqual('100');
    });
});

