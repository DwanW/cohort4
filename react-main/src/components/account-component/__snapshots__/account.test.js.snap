// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Component test should render Account Component 1`] = `ShallowWrapper {}`;
